src/pay/payment_producer.py

```diff
@@ -258,37 +265,54 @@ def try_to_pay(self, pymnt_cycle, rewards_type):
                 return True
 
             # 1- get reward data
+            reward_model = self.reward_api.get_rewards_for_cycle_map(pymnt_cycle, rewards_type)
+
+            block_reward = network_config["BLOCK_REWARD"]
+            endorsement_reward = network_config["ENDORSEMENT_REWARD"]
+
+            # 2- compute reward amount to distribute based on configuration
```

Comment:
I think we should make a compute_reward method which kinda looks like this:
```python
computed_reward_amount = compute_reward(reward_model, network_config)
```
Which then we could unit test for every possible scenario easier.